### ISSUE TITLE
UDF computations on overlapping ROIs with a common merge step

### ISSUE THREAD

**Issue by uellue** [LiberTEM/LiberTEM#651]
Discussions with https://github.com/Ptychography-4-0/ptychography/ team:

Ptychography reconstruction requires partial reconstruction and stitching of regions that are close and compact in sample coordinates and not necessarily in the "natural order" and current partitioning of the dataset, which could be a random scan with temporal ordering.

To enable distributed ptychography, it should be possible to run an UDF in such a way that data that belongs to a global ROI is passed as one data block to a reconstruction algorithm, i.e. "`process_partition()`" The local result is then merged into a global result buffer with a non-trivial merge function.

The actual reconstruction and merge can be swapped out as needed and the size and shape of regions can be adapted to the "sweet spot" of various algorithms and hardware as needed.

**Comment by uellue** [comment]
Actually, it should be possible to pass a list of ROIs to define "virtual partitions" that can be processed in parallel / arbitrary order.

**Comment by uellue** [comment]
Refs https://github.com/Ptychography-4-0/ptychography/issues/10
Refs https://github.com/Ptychography-4-0/ptychography/issues/8 for merge function that is stable for arbitrary overlapping ROIs, i.e. repeated processing
Refs https://github.com/Ptychography-4-0/ptychography/issues/12 as a simple test algorithm to validate the system

**Comment by uellue** [comment]
Just like #981, this can be solved in an "outer loop" outside of LiberTEM for now . Closing until specific needs arise!
